src/Tickets/Seating/Frontend/Session.php

```diff
@@ -340,4 +341,50 @@ public function confirm_all_reservations(): bool {
 
 		return $confirmed;
 	}
+
+	/**
+	 * Returns the seat label for a ticket in an event's registration modal/page.
```

Comment:
This returns a list, update the doc-block to reflect that.
Also: this is the doc-block of the method, not how/where it's used; I would remove the "in an event..." part.
Finally, why not just return an array of (string) seat labels from this method (as the name suggest) in place of shuttling around the whole reservation object?

Reply:
> I did that initially - then i though i was i need of the reservation id as well. So i did a small refactor. Turns out i didn't need it though, but i kept the logic because it could prove more valuble in the future to have the method return a list of all the reservation details for a specific event and ticket and the logic utilizing that to decide what's needed.

It might, and when it will be needed, it will be created. Either you need it now, or there is no point in indulging speculative engineering: add something as it _might_ be needed.